src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java

```diff
@@ -271,6 +270,19 @@ public void addConfiguration(InstanceConfiguration configuration) {
     return result;
   }
 
+  /**
+   * Choose config from list of available configs.
+   * Current implementation use round robin strategy starting at semi random element of list.
+   * Because most of times arriving requests asks for only 1 new node, we dont want 
+   * to start every time from 1 element.
+   *
+   * @param configs List of configs to choose from.
+   * @return Chosen config from list.
+   */
+  private InstanceConfiguration chooseConfigFromList(List<InstanceConfiguration> configs) {
+    return configs.get(Math.abs(configsNext++) % configs.size());
```

Comment:
Isn't this the same method as before? Or are you going to configure the value of configsNext somewhere?

Reply:
This time its static value.. so it will change with every run..

Reply:
Cool. Learned something new.